### PR TITLE
Refactoring of output functions

### PR DESCRIPTION
**Pull request by romanblanco**
One function for colored output, also removed unnecessary comments and unused functions

### CODE REVIEW

File: dottle [hunk 1]
```diff
@@ -1,106 +1,85 @@
 #!/bin/sh
 
-# DEBUG=$(printf "${DEBUG}" | tr '[:upper:]' '[:lower:]')
-
+DEBUG=false
```
Comment (ludat) [inline dottle]: This will always disable debug... before it was disabled by default (DEBUG variable was empty) and if you set it to anything (by calling dottle like `DEBUG=dsa ./dottle install`) it would activate the debug mode.

EDIT: we should change all the top level variable names to something less generic by adding a DOTTLE_ before the name

File: dottle [hunk 2]
```diff
@@ -1,106 +1,85 @@
...
+            ;;
+        error)
+            if [ ! -z "$message" ] ; then
+                printf "${RED}[error]${RESET} %b\n" "$message" 1>&2
+            else
+                printf "${RED}[error]${RESET}" 1>&2
+            fi
+            ;;
+        debug)    
+            if $DEBUG ; then
```
Comment (ludat) [inline dottle]: This will always run the content of the DEBUG env var... I really don't like that... I prefer checking if its empty

File: dottle [hunk 3]
```diff
@@ -1,106 +1,85 @@
...
+                printf "${MAGENTA}[debug]${RESET} %b\n" "$message" 1>&2
+            fi
+            ;;
+        running)    
+            printf "${YELLOW}[running]${RESET} %b" "$message"
+            ;;
+    esac
```
Comment (ludat) [inline dottle:51]: There should be a default case telling the user that we fucked up somewhere (we passed a wrong level)

File: dottle [hunk 4]
```diff
@@ -1,106 +1,85 @@
...
 
 Options:
-- None for now
+    not implemented yet
 
 Commands and args:
-- None for now
+    check             check config file
+    install           place dotfiles on place according to config file
+    --help            show this help message
```
Comment (ludat) [inline dottle]: documentation! great!

Maybe this should be in a separate PR (I'll accept it instantly)

File: dottle (hunk 2 above)
Comment (romanblanco) [inline dottle]: I see. Now I understand.
I also changed condition on [line 41](https://github.com/romanblanco/dottle/commit/a1ed73c6352783f74123b39850c41e25accdccb8#diff-b24d805ceff0cc9dc70b782d92465d3aR41) for that reason... I'll get it back, thanks.

File: dottle [hunk 5]
```diff
@@ -1,106 +1,85 @@
...
+RED=$ESC"0;31;01m"
+GREEN=$ESC"0;32;01m"
+YELLOW=$ESC"0;33;01m"
+BLUE=$ESC"0;34;01m"
+MAGENTA=$ESC"0;35;01m"
 
-running () {
-    printf "$COL_YELLOW[running]$COL_RESET %b" "$1"
+# print funcion
+output () {
```
Comment (ludat) [inline dottle:17]: The syntax in this function seems a little backwards to me.

I prefer `output "level" "message"` instead of `output "message" "level"`

File: dottle (hunk 4 above)
Comment (romanblanco) [inline dottle]: sure, it'll be better in separate PR, I've made a lot changes at once and it was quite a mess

File: dottle (hunk 2 above)
Comment (ludat) [inline dottle]: Great, I missed that one :+1:

File: dottle [hunk 6]
```diff
@@ -1,65 +1,58 @@
 #!/bin/sh
 
-# DEBUG=$(printf "${DEBUG}" | tr '[:upper:]' '[:lower:]')
-
+#DEBUG=$(printf "${DEBUG}" | tr '[:upper:]' '[:lower:]')
```
Comment (ludat) [inline dottle]: This line is useless, just delete it

File: dottle [hunk 7]
```diff
@@ -1,65 +1,58 @@
...
+YELLOW=$ESC"0;33;01m"
+BLUE=$ESC"0;34;01m"
+MAGENTA=$ESC"0;35;01m"
 
-running () {
-    printf "$COL_YELLOW[running]$COL_RESET %b" "$1"
+# print funcion
+output () {
+    # load params
+    local message=${1}
```
Comment (ludat) [inline dottle]: sadly the `local` keyword isn't part of POSIX. That's why I don't use it anywhere else. It's kind of stretchy anyway because I think every shell I've seen implements `local`... but in the repo I say "POSIX script" so better keep it up.

**Comment by romanblanco** [comment]
@ludat updated :wink:

**Comment by ludat** [comment]
Everything seems ok :D

My first accepted PR, this feels awesome
